### ISSUE TITLE
Provide guidance on how package managers should handle external dependencies.

### ISSUE THREAD

**Issue by mjlbach** [nvim-lua/nvim-package-specification#13]
Some dependencies are runtime, some are build time. The specification may want to distinguish these, as the latter could cause the plugin to fail installation.

**Comment by wbthomason** [comment]
This guidance should also include what we expect package managers to do about verifying external dependency versions.

**Comment by mjlbach** [comment]
It's hard, because we can't enforce the versioning scheme (it could be calver, not semver). I'm not sure what the best generic way to check against versions is. We could allow specifying comparator rules, regex matches on version, etc. Not sure.

**Comment by wbthomason** [comment]
Yeah, I'm not sure what we can require that isn't too onerous for plugin managers.

**Comment by lewis6991** [comment]
Personally, I think we should defer this and omit it from the initial release.
